### ISSUE TITLE
Unused parameter in mesh generators / --mesh-only does not cause warning or error

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/moose#14550]
## Bug Description
If we add a wrong parameter (non-required) in a mesh generator, the code will run fine in `--mesh-only` mode although it should produce a warning or error.

## Steps to Reproduce
Can be produced by intentionally introducing a typo in a non-required parameter.

## Impact
Prevent unexpected behavior from happening.

**Comment by friedmud** [comment]
This is about unused parameter warnings.

For this to work with `mesh-only` we would need to verify that no later Actions are registered to the current block - and only issue unused parameter warnings in that case.  Otherwise, multiple Actions reading different parameters from the same block on either side of the `mesh-only` "barrier" would generate spurious warnings

**Comment by lindsayad** [comment]
It appears that @shikhar413 is working on this? That would be awesome. I know this will highlight some errors in our mesh generator test inputs. For example in `gen_extrude.i`:
```
  [extrude]
    type = AdvancedExtruderGenerator
    input = gmg
    heights = '1 2 3'
    num_layers = '1 2 3'
    direction = '0 0 1'
    bottom_sideset = '4'
    top_sideset = '5'
    subdomain_swaps = '0 1;
                       0 2;
                       0 3'
  []
```
`bottom_sideset` and `top_sideset` are not parameters of `AdvancedExtruderGenerator`

**Comment by shikhar413** [comment]
Correct, PR is almost complete pending a few more app test fixes, since the behavior now is to error out if an unused parameter exists, which requires all existing and new input files to be free from unused parameters 

https://github.com/idaholab/moose/pull/29824

**Comment by shikhar413** [comment]
@GiudGiud @lindsayad this issue can be be closed by #29824

**Comment by GiudGiud** [comment]
closing as resolved by https://github.com/idaholab/moose/pull/29824

thanks @shikhar413 !
